src/ol/format/kmlformat.js

```diff
@@ -2390,10 +2390,10 @@ ol.format.KML.writePlacemark_ = function(node, feature, objectStack) {
     // FIXME the styles returned by the style function are supposed to be
     // resolution-independent here
     var styles = styleFunction.call(feature, 0);
-    if (styles && styles.length > 0) {
-      var style = styles[0];
+    if (styles) {
+      var style = goog.isArray(styles) ? styles[0] : styles;
```

Comment:
We can't use Array.isArray because of https://github.com/google/closure-compiler/pull/1451